File: JavaScript/String.js

var str = "ABCDEFG" // String型
str = 'ABCDEFG' // String型
str = new String("ABCDEFG") // あまり使われることはない。Object型になるため、===の利用に注意

// 文字列内の変数利用
str = `str = ${str}` // バッククォート
console.log(str)

// 一致判定
str = "ABCDEFG"
if (str == "ABCDEFG") {
    console.log("一致")
}
if (str === "ABCDEFG") {
    console.log("型も値も一致")
}

// 長さ
console.log(str.length)

// 切り出し
console.log(str.substring(2)) // CDEFG
console.log(str.substring(2, 4)) // CD

// 検索
console.log(str.indexOf("CDE")) // 2
console.log(str.includes("CDE")) // true
console.log(str.startsWith("ABC")) // true
console.log(str.endsWith("EFG")) // true
console.log(str.match(/def/i))

// 分割
str = "A,B,C,D,E"
var ary = str.split(",")
console.log(ary)

// 結合
console.log("ABC".concat("DEF", "GHI", "JKL"))

// 大文字小文字
console.log("abcde".toUpperCase())
console.log("ABCDE".toLowerCase())

// 置換
console.log("ABCDE".replace("ABC", "ZZZ"))

// 前後の空白を削除
str = " ab cd "
console.log(str.trim()) // 前後をtrim
console.log(str.trimStart()) // 前のみtrim
console.log(str.trimEnd()) // 後のみtrim

// 数値と文字列の相互変換
console.log(String(123))
console.log(parseInt("1234"))

File: Python/String.py
# coding: utf-8

# 文字列内の変数利用
print("name: %s, age: %d, isMale: %r" % ("Honda", 21, True))
print("name: {0}, age: {1}, isMale: {2}".format("Honda", 21, True))
print("name: {}, age: {}, isMale: {}".format("Honda", 21, True))

# 一致判定
s = "ABCDEFG"
if s == "ABCDEFG":
    print("一致")

# 長さ
print(len(s))

# 切り出し
print(s[1:4]) # BCD
print(s[2:]) # CDEFG
print(s[:3]) # ABC
print(s[:]) # ABCDEFG

# 検索
print("BCD" in s) # True
print(s.find("CDE")) # 2
print(s.startswith("ABC"))
print(s.endswith("EFG"))

# 分割
s = "A,B,C,D,E"
list = s.split(',')
print(list)

# 結合
print(",".join(list)) 
print("ABC" + "EFG")
print("ABC" * 3) 

# 大文字小文字
print("abc".upper())
print("ABC".lower())
print("abc".capitalize())

# 置換
s = "ABCDABCD"
print(s.replace("ABC", "ZZZ"))
print(s.replace("ABC", "ZZZ", 1)) # 置換回数1回のみ

# 前後の空白を削除
s = " ab cd "
print(s.strip())
print(s.lstrip()) # 前のみ
print(s.rstrip()) # 後ろのみ

# 数値と文字列
print(str(100))
print(int('100'))